src/model/model.js

```diff
@@ -331,9 +331,7 @@ export default class Model {
 	 *
 	 * @fires insertContent
 	 * @param {module:engine/model/documentfragment~DocumentFragment|module:engine/model/item~Item} content The content to insert.
-	 * @param {module:engine/model/selection~Selection|module:engine/model/documentselection~DocumentSelection|
-	 * module:engine/model/position~Position|module:engine/model/item~Item|
```

Comment:
This is the only place which mentions `module:engine/model/item~Item` instead of `module:engine/model/Node~Node` (which means `TextProxy` is also supported) which makes me confused about `Item` vs `Node` in `Selectable` type. Have you checked if it should be `Node` or `Item`? Because now new `Selectable` is incompatible with this line.

Reply:
Yeah - I remember that one. It boils down to that this method didn't check or call `TextNode`'s specific methods. The chain is:

* `model.insertContent()` calls `insertContent()` function
* the `insertContent()` function uses `selectable` to create a `Selection` instance (if not a `Selection` already) using `writer.createSelection()`
* `writer.createSelection()` calls `model.createSelection()` :trollface: 
* `model.createSelection()` creates new instance of `Selection`
* `Selection`'s constructors checks the `selectable` is one of `Selectable` types: (`Selection`, `DocumentSelection`, `Range`, `Position`, `Node` (!), iterable (`Array.<Range>`) or `null`.
https://github.com/ckeditor/ckeditor5-engine/blob/1649b5f1624e7e0adb43cd686d6a82b968885435/src/model/selection.js#L368-L405

So I've just unified all the usages so no `Item` nor `Element` there. Now it's all cleaned up :wastebasket: :recycle: :)

Reply:
ps.: good catch with those docs :+1: